src/skills-builder/skills-builder-modal/view-results/ViewResults.jsx

```diff
@@ -58,9 +58,24 @@ const ViewResults = () => {
       }));
 
       setJobSkillsList(jobInfo);
-      setSelectedJobTitle(jobInfo[0].name);
+      setSelectedJobTitle(results[0].name);
       setProductRecommendations(results);
       setIsLoading(false);
+      sendTrackEvent('edx.skills_builder.recommendation.shown', {
+        app_name: 'skills_builder',
+        category: 'skills_builder',
+        page: 'skills_builder',
+        selected_recommendations: {
+          job_id: results[0].id,
+          job_name: results[0].name,
+          /* We extract the title and course key into an array of objects */
+          course_keys: results[0].recommendations.course?.map(rec => ({
+            title: rec.title,
+            course_key: rec.active_run_key,
+          })),
+        },
+        is_default: true,
+      });
```

Comment:
While this may be making the code a little less DRY, I found this implementation to be a lot less clunky. Especially in terms of getting it to fire _only once_ when the recommendations first load. Also, the previous implementation was causing React to loop it's `useEffect` when it wasn't necessary, making the application a little less performant.

Moreover, all of the event code is eventually going to live somewhere else and I figure that will be the best time to refactor things further. That work is captured by [this ticket](https://2u-internal.atlassian.net/browse/APER-2392) .

Reply:
Can we put this comment in the code somewhere? This is interesting context that explains why some of the code is duplicated. I could see someone coming through and trying to refactor this and not knowing the historical context.